### PR TITLE
chore: do not publish linting settings to npm

### PR DESCRIPTION
**Pull request by adrukh**
Plugin uses eslint, update npmignore accordingly.

### CODE REVIEW

**Comment by adrukh** [comment]
Exclude `.git` from the published package too please?

**Comment by aryehbeitz** [comment]
I don't see the .git file passed. I'm pretty sure it's automatically ignored (i did a local install of the plugin and it wasn't there).
However, I would add the `dist` folder to the `.npmignore`, since npm install generates it. I noticed the error `mkdir: cannot create directory ‘dist’: File exists` when installing the plugin to a clean folder.
Your thoughts?

**Comment by adrukh** [comment]
👍  re `.git`

As for `dist`, indeed an issue - check how the cli does this and follow suit.

**Comment by aryehbeitz** [comment]
in the cli, the `package.json` has a `build` property which is apparently run on install.
I used a `postinstall` because when cloning a repo for development, the `build` is not run and I had to figure to run it manually.
I guess the proper solution is to change `postinstall` to `build`, and in the readme, add instructions for contributing: " Fork this repo, run `npm run build` and submit a PR"
Your thoughts?

**Comment by adrukh** [comment]
We need to be sure that when a user runs `npm i -g snyk`, and the php plugin is included, it can run immediately. I think that `postinstall` hooks for dependencies will run as dependencies get installed, so the `dist` folder will be created if it is triggered on `postinstall`. Not 100% sure - please validate this for both npm@3 and npm@5.

**Comment by aryehbeitz** [comment]
@adrukh 
the only way i can think of at the moment of testing this is to publish something on npm which depends on 2 projects: one with build and one with postinstall. we can then see in which the dist gets created.
Any ideas of a better way to test this?

**Comment by adrukh** [comment]
Surely we're not the only ones wondering about this. :)
Stackoverflow to the rescue?

I say try one option first, publish the plugin with it, then try `npm install` on a cloned snyk-internal repo and make sure it works. Let me know which branch this is, I'll test on a windows machine to make sure.

**Comment by aryehbeitz** [comment]
when i run `npm install` on `snyk-internal` on php branch(`feat/php_plugin`), it runs the `postinstall` for the `snyk-php-plugin` dependency (you see in the console/all versions of npm).
I guess we could test this current behavior in windows and then try changing it to `build`

**Comment by adrukh** [comment]
@gjvis @darscan does this make sense?

TL;DR: php plugin uses a minimised version of lodash which it builds on `postinstall`. Installing the cli will trigger the plugin's `postinstall` target and provide the lodash dependency.

**Comment by darscan** [comment]
BTW, using `npmignore`feels a little fragile. I way prefer the `files: ['lib', 'dist']` approach: https://docs.npmjs.com/files/package.json#files

**Comment by aryehbeitz** [comment]
@adrukh should I implement it this way?

**Comment by adrukh** [comment]
Sounds legit
